File: python-algorithms/src/valid_mountain_array.py
from typing import List

class ValidMountainArray:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3:
            return False

        last_increasing_index = 0
        last_decreasing_index = 0

        for a in range(len(arr) - 1):
            if not (arr[a] < arr[a + 1]):
                last_increasing_index = a
                break

        if last_increasing_index == 0:
            return False

        for b in range(len(arr) - 1, 0, -1):
            if not (arr[b - 1] > arr[b]):
                last_decreasing_index = b
                break

        if last_decreasing_index == len(arr) - 1:
            return False

        return last_increasing_index == last_decreasing_index

File: java-algorithms/src/main/java/com/omsprog/algorithms/lc/ValidMountainArray.java
package com.omsprog.algorithms.lc;

public class ValidMountainArray {
    public boolean validMountainArray(int[] arr) {
        int lastIncreasingIndex = 0;
        int lastDecreasingIndex = 0;

        for(int a = 0; a < arr.length -1; a++) {
            if(!(arr[a] < arr[a + 1])) {
                lastIncreasingIndex = a;
                break;
            }
        }
        if(lastIncreasingIndex == 0) return false;

        for(int b = arr.length - 1; b > 0; b--) {
            if(!(arr[b - 1] > arr[b])) {
                lastDecreasingIndex = b;
                break;
            }
        }

        if(lastDecreasingIndex == arr.length - 1) return false;

        return lastIncreasingIndex == lastDecreasingIndex;
    }
}
